app/scripts/lib/metrics.js

```diff
@@ -185,9 +187,30 @@ define(function (require, exports, module) {
         .then(sent => sent || send());
     },
 
-    _isFlushRequired (data) {
-      return data.events.length !== 0 ||
-        Object.keys(data.timers).length !== 0;
+    /**
+     * Check if a flush is required for the given `data`. A flush is
+     * required if any data has changed since the last flush.
+     *
+     * @param {Object} data - potential data to flush
+     * @param {Object} lastFlushedData - last data that was flushed.
+     * @returns {Boolean}
+     * @private
+     */
+    _isFlushRequired (data, lastFlushedData) {
+      if (! lastFlushedData) {
+        return true;
+      }
+      // Only check fields that are in the new payload. `data` could be
+      // a subset of `_lastFlushedData`, in which case no flush should occur.
+      return _.any(data, (value, key) => {
+        // these keys are distinct every flush attempt, ignore.
+        if (key === 'duration' || key === 'flushTime') {
```

Comment:
I'm OK with ignoring both `duration` and `flushTime` when doing the comparison. `_isFlushRequired` used to return `false` if both `data.events` and `data.timers` was empty, which would occur after a flush. If either array contained an entry, data would be sent. This function still acts the same, if both `events` and `timers` are empty, no send, if either has an entry, send.